src/Compilers/Test/Utilities/CSharp/CompilationTestUtils.cs

```diff
@@ -407,5 +407,48 @@ ExpressionSyntax asExpression(SyntaxNode node)
                 return expr;
             }
         }
+
+        private sealed class TopLevelNullabilityRetriever : BoundTreeWalker
```

Comment:
>TopLevelNullability [](start = 29, length = 19)

Consider renaming since this returns not just top-level nullability but types with top-level and nested nullability.